urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/response/successful/OK.java

```diff
@@ -0,0 +1,698 @@
+						   "[BasicTypeException] " + e.getMessage() );
+	    throw new ConfigurationException( e.getMessage() );
+
+	}
+    }
+
+
+    /**
+     * This method prepares OPTIONS requests.
+     *
+     * @param optionalReturnSettings An (optional) map for error retrieval and internal settings retrieval.
+     **/
+    private void prepareOPTIONS( Map<String,BasicType> optionalReturnSettings ) 
+	throws MalformedRequestException,
+	       UnsupportedVersionException,
+	       UnsupportedMethodException,
+	       UnknownMethodException,
+	       ConfigurationException,
+	       MissingResourceException,
+	       AuthorizationException,
+	       HeaderFormatException,
+	       DataFormatException,
+	       UnsupportedFormatException,
+	       SecurityException,
+	       IOException {
+
+
+	//String[] supportedMethods = this.getHTTPHandler().getSupportedMethods();
+	List<String> supportedMethods = this.getHTTPHandler().getSupportedMethods();
+	String optionsCSV = CustomUtil.implode( supportedMethods, "," );
+	
+	
+	
+	// Add default headers
+	super.addResponseHeader( "Server",            this.getHTTPHandler().getSoftwareName() ); 
+	super.addResponseHeader( "Content-Length",    "0" ); 
+	super.addResponseHeader( "Connection",        "close" );
+
+
+	// This is the actual OPTIONS response
+	super.addResponseHeader( "Allow", optionsCSV );
+	 
+    }
+
+
+    /**
+     * This method prepares OPTIONS requests.
+     *
+     * @param optionalReturnSettings An (optional) map for error retrieval and internal settings retrieval.
+     **/
+    private void prepareTRACE( Map<String,BasicType> optionalReturnSettings ) 
+	throws MalformedRequestException,
+	       UnsupportedVersionException,
+	       UnsupportedMethodException,
+	       UnknownMethodException,
+	       ConfigurationException,
+	       MissingResourceException,
+	       AuthorizationException,
+	       HeaderFormatException,
+	       DataFormatException,
+	       UnsupportedFormatException,
+	       SecurityException,
+	       IOException {
+
+
+	// "The TRACE method is used to invoke a remote, application-layer loop- back of the request message. 
+	//  The final recipient of the request SHOULD reflect the message received back to the client as the 
+	// entity-body of a 200 (OK) response."
+	// See http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html
+	
+	StringBuffer buffer = new StringBuffer();
+	Iterator<HTTPHeaderLine> iter = this.getRequestHeaders().iterator();
+	// Add all header lines to the buffer
+	//int CR = 0xD;  // 13 decimal
+	//int LF = 0xA;  // 10 decimal
+	while( iter.hasNext() ) {
+
+	    HTTPHeaderLine line = iter.next();
+	    buffer.append( line.getKey() );
+	    if( line.getValue() != null ) 
+		buffer.append( ": " ).append( line.getValue() );
+	    buffer.append( (char)Constants.CR ).append( (char)Constants.LF );
+	    
+	    
+	    //buffer.append( line.getRawBytes( CharSet  ...) );
+	}
+
+	Resource resource = new ByteArrayResource( this.getHTTPHandler(),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *resource*. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).